dipy/nn/cnn_1denoiser/cnn_1denoiser.py

```diff
@@ -0,0 +1,283 @@
+            Select optimizer. Default adam.
+        loss : floats
+            Optional list or dictionary specifying scalar coefficients to weight.
+            
+        '''
+    # Layer 1 - Convolutional Layer(16*1*16) + ReLU activation
+    #           Pooling
+    # Layer 2 - Convolutional Layer(8*16*32) + ReLU activation
+    #           Pooling
+    #           Flatten
+    #           Dense 
+        input_layer = Input(shape=(sig_length, 1))
+        x = Conv1D(filters=16, kernel_size=16, kernel_initializer='Orthogonal',
+                padding='same', name=f'Conv1')(input_layer)
+        x = Activation('relu', name=f'ReLU1')(x)
+        max_pool_1d = tf.keras.layers.MaxPooling1D(pool_size=2, strides=2, padding='valid')
+        pool1 = max_pool_1d(x)
+        x = Conv1D(filters=32, kernel_size=8, padding='same', name=f'Conv2')(pool1)
+        x = Activation('relu', name=f'ReLU2')(x)
+        pool2 = max_pool_1d(x)
+        pool2_flat = tf.keras.layers.Flatten()(pool2)
+        logits = tf.keras.layers.Dense(units=sig_length, activation='relu')(pool2_flat)
+        model = Model(inputs=input_layer, outputs=logits)
+        model.compile(optimizer=optimizer, loss=loss, metrics=metrics,loss_weights=loss_weights)
+        self.model = model
+
+    def compile(self, optimizer='adam', loss=None, metrics=None,loss_weights=None):
+        self.model.compile(optimizer=optimizer, loss=loss, metrics=metrics,
+                           loss_weights=loss_weights)
+
+    def summary(self):
+        """
+        Get the summary of the model.
+
+        The summary is textual and includes information about:
+        The layers and their order in the model.
+        The output shape of each layer.
+
+        Returns
+        -------
+        summary : NoneType
+            the summary of the model
+
+        """
+        return self.model.summary()
+    
+
+    def train_test_split(self,x=None,y=None):
+        sz = x.shape
+        x = np.reshape(x, (sz[0]*sz[1]*sz[2], sz[3]))
+        sz = y.shape
+        y = np.reshape(y, (sz[0]*sz[1]*sz[2], sz[3]))
+        return train_test_split(x,y)
+
+    def fit(self, x=None, y=None, batch_size=None, epochs=1, verbose=1,
+        callbacks=None, validation_split=0.0, validation_data=None,
+        shuffle=True, initial_epoch=0, steps_per_epoch=None,
+        validation_steps=None, validation_batch_size=None,
+        validation_freq=1, max_queue_size=10, workers=1,
+        use_multiprocessing=False):
+        """
+        Train the model on train dataset.
+
+        The fit method will train the model for a fixed
+        number of epochs (iterations) on a dataset. If given data 
+        is  4D it will convert it into 1D
+
+        Parameters
+        ----------
+        x_train         : ndarray
+            the x_train is the train dataset(noise-free data)
+        y_train         : ndarray shape=(BatchSize,)
+            the y_train is the labels of the train dataset(noise-added data)
```

Comment:
I think the word "labels" is confusing here. It usually refers to discrete values

Reply:
Which word should I use ?